src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -4038,6 +4050,13 @@ void QgsOgrProvider::open( OpenMode mode )
     }
   }
 
+  // Connect
+  if ( mode == OpenModeInitial )
```

Comment:
code indentation with the lambda is confusing. Perhaps wrap the connect() itself in { }